File: go/2415.go
package main

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func reverseOddLevels(root *TreeNode) *TreeNode {
	var helper func(a, b *TreeNode, l int)
	helper = func(a, b *TreeNode, l int) {
		if a == nil {
			return
		}
		if l%2 == 1 {
			a.Val, b.Val = b.Val, a.Val
		}
		helper(a.Left, b.Right, l+1)
		helper(a.Right, b.Left, l+1)
	}
	helper(root.Left, root.Right, 1)
	return root
}

File: python/2415.py
from typing import Optional

# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def reverseOddLevels(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        def helper(a: Optional[TreeNode], b: Optional[TreeNode], l: int):
            if a:
                if l % 2 == 1:
                    a.val, b.val = b.val, a.val
                helper(a.left, b.right, l+1)
                helper(a.right, b.left, l+1)
        helper(root.left, root.right, 1)
        return root